### ISSUE TITLE
Option to treat all strong as em when no strong is present

### ISSUE THREAD

**Issue by josteinaj** [snaekobbi/pipeline-mod-nlb#2]
It is desirable to have an option as follows: it should be possible to choose to treat all `<em>` tags in the body matter as `<strong>` when there is only `<em>` tags in the body matter.

Not sure if this is in the requirements already but it's norwegian-specific and easy to implement as a preprocessing step (XSLT) so @usama49 or I can implement this at some point for ourselves.

**Comment by bertfrees** [comment]
OK. Yes this could be added to src/main/resources/xml/block-translate.xpl. I see two possibilities:
1. A pre-processing step that changes either the element tags (em -> strong) or the style attributes.
2. Or an analysing step that determines whether strong is present and then passes a parameter to block-translate.xsl. EDIT: The analysing step could as well just be part of translate.xsl.

**Comment by josteinaj** [comment]
Right, didn't think about the CSS. I was thinking just a XSLT template that changes the elements but we'll probably need to check the CSS as well. Unless this is part of the requirements (seems not), I'll put this off until a later date.

**Comment by bertfrees** [comment]
It's part of requirement [4.3:40](http://snaekobbi.github.io/requirements/v1.0/index.xhtml#4.3:40). It's a very general one which you can fill in how you want. Priority is up to you. What exactly the option does is also up to you, it can include taking CSS into account or not. E.g. the default liblouis based translator supports `text-transform:louis-bold` etc. but you can choose to let the custom NLB translator not support that.

**Comment by josteinaj** [comment]
Ah, priority 1. We'll do it sooner rather than later then.

Yeah, we'll have to discuss what to do here at NLB. For instance, maybe only semantic emphasis and strong should be translated, which would be simpler (i.e. not `<i>`, `<b>` or CSS bold/italic). How is that done today btw? Are presentational bold/italic translated or just semantic?

**Comment by bertfrees** [comment]
There is some code in the default liblouis based translator that handles bold/italic in print CSS: see https://github.com/daisy/pipeline-mod-braille/blob/master/pipeline-braille-utils/liblouis-utils/liblouis-core/src/main/java/org/daisy/pipeline/braille/liblouis/impl/LiblouisTranslatorJnaImpl.java#L520

**Comment by josteinaj** [comment]
Interesting. I wonder what the braille experts prefer to do about presentational bold/italic/underline; personally I would assume that presentational stuff would be dropped when rendering braille.

(Just googled it and realized that all elements are supposed to be semantic, even `<i>` and `<b>`. So discarding presentational bold/italic/underline is just a matter of discarding the input CSS I suppose. In any case a preprocessing thing.)

**Comment by bertfrees** [comment]
Yup, that's right.

I've included support for presentational print CSS just in case anyone would want to use that info in the braille translation. I recall for instance that we were interested in getting text-level style information such as bold, colored, etc. from the XML suppliers, and that I proposed to use CSS for that because strong, em, i, b, u etc. are supposed to be semantic. It would be possible to use this CSS in the braille system.
